frontend/src/contexts/punchlines.tsx

```diff
@@ -1,92 +1,56 @@
 import React, {
+  Dispatch,
   PropsWithChildren,
-  useCallback,
   useContext,
-  useMemo,
```

Comment:
Any reason memo isn't being used like in some of the previous PRs?

Reply:
Ah right - so the other memoisation with other reducers (eg. in the axios hooks) was for other purposes?

Reply:
Yeah that was a slightly different setup cause it returned [state, request] where `request` was derived from `useCallback()`. These we get `const context = useReducer()` (where context == [state, dispatch]). The reason we need the `useMemo()` is whenever we zip objects into an array or object to return - even if the elements inside the array/object haven't changed React sees that you're returning a `new` array/object reference - that's why we in the axios hooks I return the result of `useMemo()`.

The difference here is that the result of `useReducer()` already takes that into consideration - & since I const unpack it (by assigning it to `const context = useReducer()` we can just return `context` & have no worries.

Reply:
Right yeah that makes sense. So it's essentially based on whether the object we're getting back needs to be memoised?

Reply:
Yep! & in this case we don't need to memoize it ourselves since `useReducer()` is returning an object that's already memoized 👌🏼